api/src/main/java/org/openmrs/validator/PersonAddressValidator.java

```diff
@@ -62,14 +62,14 @@ public void validate(Object object, Errors errors) {
 		if (log.isDebugEnabled()) {
 			log.debug(this.getClass().getName() + ".validate...");
 		}
-		
-		if (object == null) {
-			throw new IllegalArgumentException("The personAddress object should not be null");
-		}
-		
 		PersonAddress personAddress = (PersonAddress) object;
-		
-		//resolve a shorter name to display along with the error message
+//resolve a shorter name to display along with the error message
+		if (object == null) {
+			 errors.reject("error.address");
+		 } else if (!personAddress.getVoided()) {
+		// TODO - the following method should be made private in a major release
```

Comment:
@jwamalwa can you remove the TODO comments 
https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-logging